### DISCUSSION TITLE
How to present a View without a Reducer in TCA

### DISCUSSION THREAD

**Discussion by tobitech** [pointfreeco/swift-composable-architecture#2937] [Q&A]
Hello, I don't know if this has been answered before but how can I derive a `Binding` from a Store to use for navigation without scoping the store cause I want to present a View without a Reducer.

Assuming I have a Destination Reducer in my feature:
```
@Reducer
enum Destination {
    case addContact(AddContactFeature)
    case alert(AlertState<ContactsFeature.Action.Alert>)
    case docs
    case archive
}
```
I want to be able to derive a binding I can use to present views for the `docs` and `archive` destinations using the `.sheet(item:)` modifier.

The lines of code below using the sheet modifier don't compile.
```
@Perception.Bindable var store: StoreOf<ContactsFeature>
// ...
.sheet(item: $store) { _ in Text("Your Documents") }
.sheet(item: $store) { _ in Text("Your Archive") }
```

**Comment by mbrandonw** [comment]
Hi @tobitech, you can use the `.scope` operation on `$store` to derive a binding, just as you would for cases that house TCA features:

```swift
.sheet(item: $scope(state: \.destination?.docs, action: \.destination.docs) { _ in 
  Text("Your documents")
}
.sheet(item: $scope(state: \.destination?.archive, action: \.destination.archive) { _ in 
  Text("Your archive")
}
```

**Reply by mbrandonw** [reply]
Actually, this isn't quite true. I think we need to update `@Reducer` on enums to provide a `case docs(Never)` to the action. We will look into doing that soon, and then this will be a lot easier.

**Reply by tobitech** [reply]
Yeah, I tried that. Looking forward to the updates. Thanks

**Reply by stephencelis** [reply]
@tobitech Pushed a PR here if you want to take the branch for a test drive: https://github.com/pointfreeco/swift-composable-architecture/pull/2943
